### DISCUSSION TITLE
NVRAM Drive

### DISCUSSION THREAD

**Discussion by rallepalaveev** [a2stuff/a2d#742] [Ideas]
Hi, I am developing the NVRAM Drive projects and one of my friends suggested that my cards are described and recognized by A2D as "NVRAM Drive", rather than as a "Generic Block Device". Please let me know if you would like pursue this and I can provide the necessary support.

**Comment by inexorabletash** [comment]
Hey, apologies, I didn't see this. 

The logic for device identification is here: https://github.com/a2stuff/a2d/blob/ce00e066e0c5983ee88f3434e4fcf691af366d55/desk.acc/this.apple.s#L1313 - basically, if it's a ProDOS device, check if it's SmartPort and ask for device names using the SmartPort protocol. Other block devices get the generic name. 

Adding in checks for specific device types is possible, given reliable ID bytes.

**Comment by inexorabletash** [comment]
Okay, done in c51136e7e004db55bdee94aeebe522adde11fa51 if I got the signature correct.
